src/lsst/cmservice/web_app/static/js/script-grid-utils.js

```diff
@@ -0,0 +1,192 @@
+// list of statuses to reset to if current status is failed or rejected
+const resetFromRejected = {
+    WAITING: 0,
+    READY: 1,
+    PREPARED: 2
+};
+// list of statuses to reset to if current status is reviewable
+const resetFromReviewable = {
+    ACCEPTED: 5,
+    REJECTED: -3,
+};
+
+const updateStatus = () => {
+    const rowIndex = parseInt(document.querySelector("#rowIndex").innerText);
+    const newStatus = parseInt(document.querySelector("#targetStatus").value);
+    const fullname = String(document.querySelector("#scriptFullname").innerText)
+    const scriptId = scriptGridApi.getRowNode(rowIndex).data.id;
+
+    let url;
+    let resetJson = null;
+
+    // accept script
+    if(newStatus === 5){
+        url = `/cm-service/v1/script/action/${scriptId}/accept`;
+    }
+    // reject script
+    else if(newStatus === -3){
+        url = `/cm-service/v1/script/action/${scriptId}/reject`;
+    }
+    // reset script
+    else {
+        let resetObj = {
+            fullname: fullname,
+            status: newStatus,
+        };
+
+        resetJson = JSON.stringify(resetObj);
+        url = reset_script_endpoint;
+    }
+
+    // send request
+    let xhr = new XMLHttpRequest();
+    xhr.open('POST', url, true);
+    xhr.setRequestHeader('Content-Type', 'application/json; charset=UTF-8');
+    xhr.onload = function () {
+        if (xhr.status === 201) {
+            // update script status in grid
+            scriptGridApi
+                .getRowNode(rowIndex)
+                .setDataValue("status", document.querySelector("#targetStatus").selectedOptions[0].text);
+            // refresh grid cells on client side to disable/enable reset button according to new status
+            scriptGridApi.refreshCells({
+                force: true,
+                suppressFlash: false,
+              });
+            resetScriptModal.close();
+        } else {
+            resetScriptModal.close();
+            errorModal.showModal();
+            errorMessage.innerText = JSON.parse(xhr.responseText).detail;
+        }
+    };
+    xhr.onerror = function () {
+        console.error("Error:", xhr.status, xhr.responseText);
+        resetScriptModal.close();
+        errorModal.showModal();
+        errorMessage.innerText = xhr.responseText;
+    };
+    xhr.send(resetJson);
```

Comment:
Is there a reason not to use Fetch here instead of XMLHttpRequest?

Reply:
I'll try fetch